org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/utils/DOMUtils.java

```diff
@@ -89,21 +91,56 @@ public static boolean isXSD(String uri) {
 	 * @return true if the XML document is a RelaxNG grammar and false otherwise.
 	 */
 	public static boolean isRelaxNG(DOMDocument document) {
+		if (isRelaxNGXMLSyntax(document)) {
+			return true;
+		}
 		if (document == null) {
 			return false;
 		}
 		String uri = document.getDocumentURI();
 		return isRelaxNG(uri);
 	}
 
+	/**
+	 * Returns true if the XML document is a RelaxNG grammar XML syntax and false
+	 * otherwise.
+	 * 
+	 * @return true if the XML document is a RelaxNG grammar XML syntax and false
+	 *         otherwise.
+	 */
+	public static boolean isRelaxNGXMLSyntax(DOMDocument document) {
+		if (document == null) {
+			return false;
+		}
+		String uri = document.getDocumentURI();
+		if (isRelaxNGXMLSyntax(uri)) {
+			return true;
+		}
+		// check root element is bound with RNG namespace
+		// (http://relaxng.org/ns/structure/1.0)
+		return checkRootNamespace(document, RELAXNG_GRAMMAR_NS);
+	}
+
+	/**
+	 * Returns true if the given URI is a RelaxNG grammar XML syntax and false
+	 * otherwise.
+	 * 
+	 * @param uri the URI to check
+	 * @return true if the given URI is a RelaxNG grammar XML syntax and false
+	 *         otherwise.
+	 */
+	public static boolean isRelaxNGXMLSyntax(String uri) {

```

Comment:
I think it would be a good idea to rename this function to incorporate "uri" in the name, so it's obvious that this doesn't try to check the namespace like the other `isRelaxNGXMLSyntax does

Reply:
What about `isRelaxNGUriXMLSyntax` ?

Reply:
fixed